src/main/java/org/zeroturnaround/javarebel/maven/GenerateRebelMojo.java

```diff
@@ -308,6 +324,50 @@ else if (POM_PACKAGING.equals(packaging)) {
     }
   }
 
+  /**
+   * Generates rebel-remote.xml.
+   * @throws MojoExecutionException
+   */
+  private void generateRebelRemoteXmlFile(File rebelRemoteXmlFile) throws MojoExecutionException {
+    getLog().info("Generating rebel-remote.xml on : " + rebelRemoteXmlFile.getAbsolutePath());
+
+    Writer w = null;
+    if (this.showGenerated) {
+      try {
+        w = new StringWriter();
+        generateRebelRemoteXml(w);
+        getLog().info(w.toString());
+      } catch (IOException e) {
+        getLog().debug("Detected exception during 'showGenerated' : ", e);
+      }
+    }
+
+    try {
+      rebelXmlDirectory.mkdirs();
+      w = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(rebelRemoteXmlFile), CHARTSET_UTF8));
```

Comment:
Is there any particular reason why the previous invocation result of generateRebelRemoteXml can not be used when writing the actual file?

Reply:
`generateRebelRemoteXml` is currently written to take in a `Writer` in which the `rebel-remote.xml` contents are written into. Those two `generateRebelRemoteXml` invocations take in different types of `Writer` instances depending where its written.